discovery/mdns.md

```diff
@@ -0,0 +1,94 @@
+# Multicast DNS
+Author: Richard Schneider (makaretu@gmail.com)
+
+
+## Overview
+
+The goal is to allow peers to discover each other when on the same local network with zero configuration. 
+MDNS uses a multicast system of DNS records; this allows all peers on the local network to see all query responses.
+
+Conceptually, it is very simple.  When a peer starts (or detects a network change), it sends a query for all peers. 
+As responses come in, the peer adds the other peers information into is local database of peers.
+
+## Definitions
+
+`service-name` is the DNS-SD service name for all IPFS peers. It is defined as `_ipfs._udp.local`.
+
+`host-name` is the name of the peer.  It is derived from the peer's ID and `ipfs.local`, for example 
+`Qmid.ipfs.local`.
+
+`peer-id` is the ID of the peer.  It normally is the base-58 enconding of the hash of the peer's public key.
+
+`port` is a port that the peer listens on. Normally 4001.
+
+## Peer Discovery
+
+### Request
+
+To find all peers, a DNS message is sent with the question `_ipfs._udp.local PTR`. 

```

Comment:
We'll probably want to announce multiple services. We should at least default to announcing `_libp2p` (and only announce `_ipfs` if configured to do so).

Reply:
I'm happy with whatever the community decides for the name.  My preference is `_p2p`.

Reply:
SGTM.